--- a/dhpython/build/plugin_flit.py
+++ b/dhpython/build/plugin_flit.py
@@ -27,9 +27,13 @@
 import os
 import os.path as osp
 import shutil
+import sys
 import sysconfig
 try:
-    import tomli
+    if sys.version_info >= (3, 11):
+        import tomllib
+    else:
+        import tomli as tomllib
 except ModuleNotFoundError:
     # Plugin still works, only needed for autodetection
     pass
--- a/dhpython/build/plugin_pyproject.py
+++ b/dhpython/build/plugin_pyproject.py
@@ -24,9 +24,13 @@
 import logging
 import os.path as osp
 import shutil
+import sys
 import sysconfig
 try:
-    import tomli
+    if sys.version_info >= (3, 11):
+        import tomllib
+    else:
+        import tomli as tomllib
 except ModuleNotFoundError:
     # Plugin still works, only needed for autodetection
     pass
@@ -66,7 +70,7 @@
 
         try:
             with open('pyproject.toml', 'rb') as f:
-                pyproject = tomli.load(f)
+                pyproject = tomllib.load(f)
             if pyproject.get('build-system', {}).get('build-backend'):
                 result += 10
             else:
